ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/JcsSignerSpec.scala

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.meteringreport
+
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Time.Timestamp
+import com.daml.platform.apiserver.meteringreport.JcsSigner.VerificationStatus._
+import com.daml.platform.apiserver.meteringreport.MeteringReport.{
+  ApplicationReport,
+  Check,
+  ParticipantReport,
+  Request,
+  Scheme,
+}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+
+class JcsSignerSpec extends AnyWordSpec with Matchers {
+
+  import HmacSha256._
+
+  private val scheme: Scheme = "test"
+  private val hmacKey: Key = HmacSha256.generateKey()
```

Comment:
It might be dangerous to use randomly generated key for a test, could be a source for flakiness?

Reply:
actually should be fine for this use case

Reply:
I use a combination of random and static keys.